nncf/tensorflow/pruning/base_algorithm.py

```diff
@@ -356,3 +356,10 @@ def _calculate_pruned_layers_summary(self) -> List[PrunedLayerSummary]:
 
     def strip_model(self, model: tf.keras.Model) -> tf.keras.Model:
         return strip_model_from_masks(model, self._op_names)
+
+    def prepare_for_inference(self, make_model_copy: bool = True) -> tf.keras.Model:
+        # Pruning transformer create new copy of the model.
+        model = strip_model_from_masks(self.model, self._op_names)
```

Comment:
Are you sure that the weights are copied?

Reply:
Yes